### PR TITLE
feat(cmd): extend workspace commands with settings, security, and billing

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

Add workspace settings, security, and billing commands to the CLI. Introduces `config.IsSaaS()` as a package-level function to replace scattered `AccessToken` checks, applying consistent SaaS/OnPrem guards across all relevant commands.

## New Commands

| Command | Description | Availability |
|---------|-------------|--------------|
| `workspace preferences [update]` | Workspace preferences | OnPrem + SaaS |
| `workspace notifications [update]` | Notification settings | OnPrem + SaaS |
| `workspace access-control [update]` | Access control settings | OnPrem + SaaS |
| `workspace authentication [update]` | Auth/MFA settings | SaaS only |
| `workspace mfa-methods` | Allowed MFA methods | SaaS only |
| `workspace usage` | Billing estimate | SaaS only |

## Changes

- **`config.IsSaaS()`**: package-level function in `config/config.go` that replaces scattered `AccessToken != ""` checks for deployment type detection
- **Consistent error messages**: OnPrem → `"self-hosted workspaces"`, SaaS → `"Alpacon Cloud workspaces"`
- **Non-interactive guard**: all `*_update` commands now fail fast with a clear error when not running in an interactive terminal
- **MFA retry flow**: `authentication update` and `access-control update` open browser for MFA, poll for completion, and automatically retry the operation
- **Unit tests**: added for all new API functions (`httptest`-based)

### Universal
- [ ] Code follows project style guidelines
- [ ] Self-reviewed the code
- [ ] No hardcoded secrets or credentials
- [ ] Updated relevant documentation

### Go
- [ ] Error handling is complete
- [ ] Tests include edge cases
- [ ] `go mod tidy` executed

### CODE REVIEW

File: api/workspace/usage.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		} `json:"results"`
+	}
+	if err := json.Unmarshal(body, &response); err != nil {
+		return "", fmt.Errorf("failed to parse workspace list: %w", err)
+	}
+
+	for _, ws := range response.Results {
+		if ws.SchemaName == workspaceName {
+			return ws.ID, nil
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/workspace/usage.go:70]: Fixed. Added a pagination loop that follows the `next` field in the response. Relative next URLs are resolved against the current URL before each subsequent request.

File: api/mfa/mfa.go [hunk 2]
```diff
@@ -73,6 +73,25 @@ func GetMFALinkForSudo(ac *client.AlpaconClient, serverName string) (string, err
...
+	}
+	responseBody, err := ac.SendGetRequest(utils.BuildURL(mfaURL, "", params))
+	if err != nil {
+		return "", fmt.Errorf("failed to get the MFA URL: %w", err)
+	}
+
+	var mfaResp mfaResponse
+	_ = json.Unmarshal(responseBody, &mfaResp)
+
+	return mfaResp.MfaURL, nil
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/mfa/mfa.go:97]: Fixed. Now propagating the `json.Unmarshal` error and returning an explicit error when `MfaURL` is empty, so callers fail fast with a clear message instead of opening an empty URL.

File: cmd/workspace/workspace_usage.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	alpacon ws usage`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg, err := config.LoadConfig()
+		if err != nil {
+			utils.CliErrorWithExit("Not logged in. Run 'alpacon login' first.")
+		}
+		isSaaS, err := config.IsSaaS()
+		if err != nil {
+			utils.CliErrorWithExit("Not logged in. Run 'alpacon login' first.")
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_usage.go]: Fixed. Replaced the redundant `config.IsSaaS()` call (which re-reads config from disk) with `cfg.AccessToken == ""` since `cfg` is already loaded just above.

File: cmd/workspace/workspace_authentication_update.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		}
+
+		cfg, err := config.LoadConfig()
+		if err != nil {
+			utils.CliErrorWithExit("Not logged in. Run 'alpacon login' first.")
+		}
+		isSaaS, err := config.IsSaaS()
+		if err != nil {
+			utils.CliErrorWithExit("Not logged in. Run 'alpacon login' first.")
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_authentication_update.go]: Fixed. Replaced the redundant `config.IsSaaS()` call with `cfg.AccessToken == ""` since `cfg` is already loaded just above.

File: cmd/workspace/workspace_preferences_update.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	Use:   "update",
+	Short: "Update workspace preferences",
+	Long: `Update workspace preferences by opening the current settings in your editor.
+Modify the desired fields, save, and close the editor to apply changes.`,
+	Example: `
+	alpacon workspace preferences update
+	alpacon ws preferences update`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if !utils.IsInteractiveShell() {
+			utils.CliErrorWithExit("this command requires an interactive terminal")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_preferences_update.go]: Fixed. Changed to "This command requires an interactive terminal."

File: cmd/workspace/workspace_notifications_update.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	Use:   "update",
+	Short: "Update workspace notification settings",
+	Long: `Update workspace notification settings by opening the current settings in your editor.
+Modify the desired fields, save, and close the editor to apply changes.`,
+	Example: `
+	alpacon workspace notifications update
+	alpacon ws noti update`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if !utils.IsInteractiveShell() {
+			utils.CliErrorWithExit("this command requires an interactive terminal")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_notifications_update.go]: Fixed. Changed to "This command requires an interactive terminal."

File: cmd/workspace/workspace_access_control_update.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	Use:   "update",
+	Short: "Update workspace access control settings",
+	Long: `Update workspace access control settings by opening the current settings in your editor.
+Modify the desired fields, save, and close the editor to apply changes.`,
+	Example: `
+	alpacon workspace access-control update
+	alpacon ws acl update`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if !utils.IsInteractiveShell() {
+			utils.CliErrorWithExit("this command requires an interactive terminal")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_access_control_update.go]: Fixed. Changed to "This command requires an interactive terminal."

File: cmd/workspace/workspace_authentication_update.go [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	Use:   "update",
+	Short: "Update workspace authentication settings",
+	Long: `Update workspace authentication settings by opening the current settings in your editor.
+Modify the desired fields, save, and close the editor to apply changes.`,
+	Example: `
+	alpacon workspace authentication update
+	alpacon ws auth update`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if !utils.IsInteractiveShell() {
+			utils.CliErrorWithExit("this command requires an interactive terminal")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_authentication_update.go]: Fixed. Changed to "This command requires an interactive terminal."

File: api/workspace/preferences.go [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+		return nil, fmt.Errorf("failed to retrieve workspace preferences: %w", err)
+	}
+
+	data, err := utils.ProcessEditedData(responseBody)
+	if err != nil {
+		return nil, err
+	}
+
+	responseBody, err = ac.SendPatchRequest(preferencesURL, data)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/workspace/preferences.go:37]: The current implementation is appropriate because:
- `EditAuthentication` in the same package (`api/workspace/authentication.go`) follows the same GET→editor→return structure and has no unit tests either.
- `utils.ProcessEditedData` is an OS-level operation that spawns `$EDITOR` and connects `os.Stdin`. HTTP mocking alone cannot bypass the editor step. Additionally, when no changes are detected, it calls `os.Exit` directly, which would terminate the test process.
- Testing this function requires adding test-only hooks (e.g., a package-level var) to production code, which is inconsistent with the existing patterns in this package.

Editor-interactive update functions are treated as integration-test concerns in this package.

File: api/workspace/notifications.go [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+		return nil, fmt.Errorf("failed to retrieve notification settings: %w", err)
+	}
+
+	data, err := utils.ProcessEditedData(responseBody)
+	if err != nil {
+		return nil, err
+	}
+
+	responseBody, err = ac.SendPatchRequest(notificationsURL, data)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/workspace/notifications.go:37]: The current implementation is appropriate because:
- `EditAuthentication` in the same package (`api/workspace/authentication.go`) follows the same GET→editor→return structure and has no unit tests either.
- `utils.ProcessEditedData` is an OS-level operation that spawns `$EDITOR` and connects `os.Stdin`. HTTP mocking alone cannot bypass the editor step. Additionally, when no changes are detected, it calls `os.Exit` directly, which would terminate the test process.
- Testing this function requires adding test-only hooks (e.g., a package-level var) to production code, which is inconsistent with the existing patterns in this package.

Editor-interactive update functions are treated as integration-test concerns in this package.

File: cmd/workspace/workspace_usage.go [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+		if !isSaaS {
+			utils.CliErrorWithExit("This command is only available on Alpacon Cloud workspaces.")
+		}
+
+		cfg, err := config.LoadConfig()
+		if err != nil {
+			utils.CliErrorWithExit("Failed to load configuration: %s.", err)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/workspace/workspace_usage.go]: Fixed. Added `IsSaaS()` as a method on the `Config` struct and updated commands that also call `config.LoadConfig()` (this file and `workspace_authentication_update.go`) to load config once and derive the SaaS check from the already-loaded value, eliminating the redundant I/O.

File: api/workspace/usage.go [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+			if ws.SchemaName == workspaceName {
+				return ws.ID, nil
+			}
+		}
+
+		if response.Next == nil {
+			break
+		}
+		page = *response.Next
+	}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/workspace/usage.go]: Fixed. Added `|| *response.Next == 0` to the break condition to prevent an infinite loop when the API returns `next: 0` instead of `null`.

**Review by eunyoung14 — APPROVED** [review]
